airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java

```diff
@@ -111,8 +113,8 @@ public AirbyteConnectionStatus check(JsonNode config) {
     try {
       final String datasetId = config.get(CONFIG_DATASET_ID).asText();
       final BigQuery bigquery = getBigQuery(config);
-
-      if (!bigquery.getDataset(datasetId).exists()) {
```

Comment:
should we just add a `createDatasetIfDoesNotExistHelper`? looks like the exact same code here and on line 221.